source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -533,12 +492,133 @@ namespace AdaptiveCards::Rendering::Uwp
                         m_copyStreamOperations.push_back(copyStreamOperation);
 
                         copyStreamOperation.Completed(
```

Comment:
>copyStreamOperation [](http://example.com/codeflow?start=24&length=19)

[preexisting] The completed callback pattern is hard to follow compared to coroutines.  Are there any plans to begin using `co_await` instead?

Reply:
I'm not sure if we've discussed this refactor. I agree that the callbacks are difficult to follow, especially since there are so many in this class.

Maybe I can file a ticket for refactoring since this is likely an issue in other files as well?

Reply:
I filed an issue on the repo: https://github.com/microsoft/AdaptiveCards/issues/8304